near-sdk/src/json_types/public_key.rs

```diff
@@ -67,17 +80,19 @@ impl Base58PublicKey {
 
 impl From<Base58PublicKey> for Vec<u8> {
     fn from(v: Base58PublicKey) -> Vec<u8> {
-        v.0
+        let mut out = vec![v.curve as u8];
+        out.extend(v.data);
+        out
```

Comment:
```suggestion
        v.data.insert(0, v.curve as u8);
        v.data
```

Aside from the other comment, can re-use the allocation since this consumes the public key. (also have to make v `mut`)